test/comp_player_test.exs

```diff
@@ -0,0 +1,141 @@
+defmodule Comp_player_test do
+  use ExUnit.Case
+  doctest Comp_player
+
+  describe "when board is empty" do
+    setup do
+      %{
+        initial_state: %State{
+          board: %{
+            {0, 0} => 1,
+            {0, 1} => 2,
+            {0, 2} => 3,
+            {1, 0} => 4,
+            {1, 1} => 5,
+            {1, 2} => 6,
+            {2, 0} => 7,
+            {2, 1} => 8,
+            {2, 2} => 9
+          },
+          player: "X"
+        }
+      }
+    end
+
+    test "finds all open spaces", context do
+      assert Comp_player.available_moves(context[:initial_state]) == %{
+               {0, 0} => 1,
+               {0, 1} => 2,
+               {0, 2} => 3,
+               {1, 0} => 4,
+               {1, 1} => 5,
+               {1, 2} => 6,
+               {2, 0} => 7,
+               {2, 1} => 8,
+               {2, 2} => 9
+             }
+    end
+
+    test "returns a random key of an open space", context do
+      move = Comp_player.select_move(context[:initial_state])
+      assert Map.fetch!(Comp_player.available_moves(context[:initial_state]), move)
+    end
+  end
+
+  describe "when board has one move" do
+    setup do
+      %{
+        one_move_state: %State{
+          board: %{
+            {0, 0} => "X",
+            {0, 1} => 2,
+            {0, 2} => 3,
+            {1, 0} => 4,
+            {1, 1} => 5,
+            {1, 2} => 6,
+            {2, 0} => 7,
+            {2, 1} => 8,
+            {2, 2} => 9
+          },
+          player: "X"
+        }
+      }
+    end
+
+    test "finds all open spaces", context do
+      assert Comp_player.available_moves(context[:one_move_state]) == %{
+               {0, 1} => 2,
+               {0, 2} => 3,
+               {1, 0} => 4,
+               {1, 1} => 5,
+               {1, 2} => 6,
+               {2, 0} => 7,
+               {2, 1} => 8,
+               {2, 2} => 9
+             }
+    end
+
+    test "returns a random key of an open space", context do
+      move = Comp_player.select_move(context[:one_move_state])
+      assert Map.fetch!(Comp_player.available_moves(context[:one_move_state]), move)
+    end
+  end
+
+  describe "when board is nearly full" do
+    setup do
+      %{
+        nearly_full_board_state: %State{
+          board: %{
+            {0, 0} => "X",
+            {0, 1} => "O",
+            {0, 2} => "X",
+            {1, 0} => "O",
+            {1, 1} => "X",
+            {1, 2} => "O",
+            {2, 0} => "X",
+            {2, 1} => "O",
+            {2, 2} => 9
+          },
+          player: "X"
+        }
+      }
+    end
+
+    test "finds the only open space", context do
+      assert Comp_player.available_moves(context[:nearly_full_board_state]) == %{{2, 2} => 9}
+    end
+
+    test "returns key of the only open space", context do
+      assert Comp_player.select_move(context[:nearly_full_board_state]) == {2, 2}
```

Comment:
What is the elixir casing convention, here?  Is it `StartCase` or `Snake_Case`?  It won't necessarily harm your code to split off from a language's conventions, but it's usually encouraged to find the language-level idiom for casing and naming, and stick to that (especially for public interfaces)